### PR TITLE
Integrate user management in person

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Moves the user management to the person and makes the user management panel advanced mode only.

(still need to do some translations)

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/person/dialog-person-detail.ts [hunk 1]
```diff
@@ -64,9 +84,15 @@ class DialogPersonDetail extends LitElement {
       this._userId = this._params.entry.user_id || undefined;
       this._deviceTrackers = this._params.entry.device_trackers || [];
       this._picture = this._params.entry.picture || null;
+      this._user = this._userId
+        ? this._params.users.find((user) => user.id === this._userId)
+        : undefined;
+      this._isAdmin = this._user?.group_ids[0] === SYSTEM_GROUP_ID_ADMIN;
```
Comment (ludeeus) [inline src/panels/config/person/dialog-person-detail.ts:90]:
```suggestion
      this._isAdmin = this._user?.group_ids.includes(SYSTEM_GROUP_ID_ADMIN);
```

File: src/panels/config/person/dialog-person-detail.ts [hunk 2]
```diff
@@ -115,15 +141,33 @@ class DialogPersonDetail extends LitElement {
...
-              )}"
-              .hass=${this.hass}
-              .value=${this._userId}
-              .users=${this._params.users}
-              @value-changed=${this._userChanged}
-            ></ha-user-picker>
+            <ha-formfield label="Allow person to login">
+              <ha-switch
+                @change=${this._allowLoginChanged}
+                .disabled=${this._user &&
```
Comment (ludeeus) [inline src/panels/config/person/dialog-person-detail.ts:147]: Should admins be able to change this for other admins?

File: src/panels/config/person/dialog-person-detail.ts [hunk 3]
```diff
@@ -213,9 +268,39 @@ class DialogPersonDetail extends LitElement {
...
+    const target = ev.target;
+    if (target.checked) {
+      target.checked = false;
+      showAddUserDialog(this, {
+        userAddedCallback: async (user?: User) => {
+          if (user) {
+            target.checked = true;
+            this._user = user;
+            this._userId = user.id;
+            this._isAdmin = user.group_ids[0] === SYSTEM_GROUP_ID_ADMIN;
```
Comment (ludeeus) [inline src/panels/config/person/dialog-person-detail.ts:285]:
```suggestion
            this._isAdmin = user.group_ids.includes(SYSTEM_GROUP_ID_ADMIN);
```

File: src/panels/config/person/dialog-person-detail.ts [hunk 4]
```diff
@@ -213,9 +268,39 @@ class DialogPersonDetail extends LitElement {
...
+            this._params?.refreshUsers();
+          }
+        },
+        name: this._name,
+      });
+    } else if (this._userId) {
+      if (
+        !(await showConfirmationDialog(this, {
+          text: "Are you sure you want to remove this user?",
```
Comment (ludeeus) [inline src/panels/config/person/dialog-person-detail.ts:294]: localize?
```suggestion
          text: "Are you sure you want to remove this user?",
          confirmText: "remove",
          dismissText: "cancel"
```

File: src/panels/config/person/dialog-person-detail.ts [hunk 5]
```diff
@@ -228,9 +313,70 @@ class DialogPersonDetail extends LitElement {
...
+  private async _changePassword() {
+    if (!this._user) {
+      return;
+    }
+    const credential = this._user.credentials.find(
+      (cred) => cred.type === "homeassistant"
+    );
+    if (!credential) {
+      showAlertDialog(this, {
+        title: "No Home Assistant credentials found.",
```
Comment (ludeeus) [inline src/panels/config/person/dialog-person-detail.ts:325]: localize?

File: src/panels/config/person/dialog-person-detail.ts [hunk 6]
```diff
@@ -228,9 +313,70 @@ class DialogPersonDetail extends LitElement {
...
+      ),
+    });
+  }
+
   private async _updateEntry() {
     this._submitting = true;
     try {
+      if (
+        (this._userId && this._name !== this._params!.entry?.name) ||
+        this._isAdmin !== (this._user?.group_ids[0] === SYSTEM_GROUP_ID_ADMIN)
```
Comment (ludeeus) [inline src/panels/config/person/dialog-person-detail.ts:370]:
```suggestion
        this._isAdmin !== (this._user?.group_ids.includes(SYSTEM_GROUP_ID_ADMIN))
```

File: src/panels/config/person/dialog-person-detail.ts (hunk 5 above)
Comment (ludeeus) [reply src/panels/config/person/dialog-person-detail.ts:325]: Didn't read your description.
> (still need to do some translations)

File: src/panels/config/person/dialog-person-detail.ts (hunk 2 above)
Comment (bramkragten) [reply src/panels/config/person/dialog-person-detail.ts:147]: Yes

File: src/panels/config/person/dialog-person-detail.ts [hunk 7]
```diff
@@ -228,9 +325,70 @@ class DialogPersonDetail extends LitElement {
...
+  private async _changePassword() {
+    if (!this._user) {
+      return;
+    }
+    const credential = this._user.credentials.find(
+      (cred) => cred.type === "homeassistant"
+    );
+    if (!credential) {
+      showAlertDialog(this, {
+        title: "No Home Assistant credentials found.",
```
Comment (ludeeus) [inline src/panels/config/person/dialog-person-detail.ts:337]: I think this is the only one missing localize now

**Review by ludeeus — APPROVED** [review]

File: src/panels/config/person/dialog-person-detail.ts (hunk 7 above)
Comment (bramkragten) [reply src/panels/config/person/dialog-person-detail.ts:337]: Should not happen anyway 😄

**Pull request merged**